viewer/packages/data-providers/src/image-360-data-providers/CoreDm360ImageAnnotationProvider.ts

```diff
@@ -54,15 +54,25 @@ export class CoreDm360ImageAnnotationProvider implements Image360AnnotationProvi
               isSameImage360RevisionId(revision.identifier, foundRevisionId)
             ) !== undefined
         );
-      return (await Promise.all(revisions.map(revision => getAnnotationInfoForRevision(entity, revision)))).flat();
+      return (
+        await Promise.all(
+          revisions.map(revision => getAnnotationInfoForRevision(entity, revision, relatedRevisionsAndAnnotations))
+        )
+      ).flat();
     }
 
     async function getAnnotationInfoForRevision(
       entity: Image360Entity<DMDataSourceType>,
-      revision: Image360RevisionEntity<DMDataSourceType>
+      revision: Image360RevisionEntity<DMDataSourceType>,
+      relatedRevisionsAndAnnotations: Image360AnnotationsForInstanceResult
     ): Promise<Image360AnnotationAssetQueryResult<DMDataSourceType>[]> {
       const annotations = await revision.getAnnotations();
-      return annotations.map(annotation => ({
+      const filteredAnnotations = annotations.filter(annotation =>
+        relatedRevisionsAndAnnotations.annotationIds.find(relatedRevisionAnnotationId =>
+          isSameDMIdentifier(relatedRevisionAnnotationId, annotation.annotation.annotationIdentifier)
+        )
```

Comment:
Maybe we should use `some` instead of `find`? I think it's about equivalent here, but `find` returns the found object, while `some` returns a boolean which says whether there is an object that satisfies the predicate